### ISSUE TITLE
IndentationCheck fails for chained method calls

### ISSUE THREAD

**Issue by kukreja-vikramaditya** [checkstyle/checkstyle#4070]
```sh
/var/tmp $ javac Main.java
```
```java
/var/tmp $ cat Main.java

import java.nio.file.Path;
public class Main { 
    void foo(Path destpath, Path sourcePath) { 
        destpath
.resolve(sourcePath.subpath(0,1)) //indent:0 exp:12 warn
            .resolve(sourcePath.subpath(0,1));
    }
} 
```
```xml
/var/tmp $ cat config.xml

<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
    <module name="Indentation">
      <property name="basicOffset" value="4"/>
      <property name="braceAdjustment" value="0"/>
      <property name="caseIndent" value="4"/>
      <property name="throwsIndent" value="8"/>
    </module>
    </module>
</module>
```
```sh
/var/tmp $ java -jar checkstyle-7.6-all.jar -c config.xml Main.java

Starting audit...
Audit done.
```
---------------

Expected error for L5 as the indentation should have been 12. This problem seems to be line with #3802 . The code for [MethodCallLineWrap](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/MethodCallHandler.java) is not able to handle complex ast structures. Currently it only requires these children in order
`SLIST`
`EXPR`
`METHOD_CALL`

--------------

**Comment by rnveach** [comment]
No violation:
````
        destpath
.resolve(sourcePath.subpath(0,1))
            .resolve(sourcePath.subpath(0,1));
````
No violation:
````
        destPath = destinationPath
.resolve(sourcePath.subpath(0));
````
Violation:
````
        destpath
.resolve(sourcePath.subpath(0,1));
````

I agree, something is definitely wrong. I am familiar with this check, but there are still things I haven't dived into.

> I have 2 approach in mind 

A rewrite will be a long and tedious process. I don't know if we have time during GSoC.
Fixing different cases may lead us to the same destination or may make the code unmanageable more than it already is.

If we are to do point 2, I think we should make a plan first.
This might go beyond method calls as we need to be able to support nested statements. https://github.com/checkstyle/checkstyle/issues/3802#issuecomment-287621315 
They shouldn't be required to all be at the same indentation.

**Comment by rnveach** [comment]
@boyravikumar The issue is open so it has not been resolved. It is not even approved and there is no thoughts currently on how to fix this.

**Comment by Stefanqn** [comment]
yeah, it's kind of sad 
```java
x.map(Object::toString)
    .map(Object::toString)
  .map(Object::toString)
   .map(Object::toString)
```
can't be validated

**Comment by Vyom-Yadav** [comment]
I am on it.
